cmd/nerdctl/builder.go

```diff
@@ -58,29 +58,83 @@ func newBuilderPruneCommand() *cobra.Command {
 	}
 
 	AddStringFlag(buildPruneCommand, "buildkit-host", nil, "", "BUILDKIT_HOST", "BuildKit address")
+
+	buildPruneCommand.Flags().BoolP("all", "a", false, "Remove all unused build cache, not just dangling ones")
+	buildPruneCommand.Flags().BoolP("force", "f", false, "Do not prompt for confirmation")
 	return buildPruneCommand
 }
 
 func builderPruneAction(cmd *cobra.Command, _ []string) error {
-	globalOptions, err := processRootCmdFlags(cmd)
+	options, err := processBuilderPruneOptions(cmd)
 	if err != nil {
 		return err
 	}
-	buildkitHost, err := getBuildkitHost(cmd, globalOptions.Namespace)
+
+	if !options.Force {
+		var (
+			confirm string
+			msg     string
+		)
+
+		if !options.All {
```

Comment:
nit: might be slightly easier to read if you reverse these conditionals since we need to handle both cases anyway